packages/nimble-components/src/select/specs/filerable-select-hld.md

```diff
@@ -0,0 +1,78 @@
+# Filterable Select
+
+## Problem Statement
+
+Our clients have a need of a filterable dropdown component that does not require arbitrary text as a value, but only values available in the dropdown. The filtering will have the following behaviors:
+
+-   The filter text will match _any_ text within the `textContent` of each `ListOption`
+-   Each time the dropdown is opened the filter text is cleared
+-   While the dropdown is opened all keystrokes except `<ArrowUp>` and `<ArrowDown>` will apply to the filter.
+-   Pressing `<Esc>` will close the dropdown and revert the value to what it was prior to opening dropdown.
+
+## Links To Relevant Work Items and Reference Material
+
+-   [#1273: Filterable NimbleSelect](https://github.com/ni/nimble/issues/1273)
+-   [Filterable Select IxD](./IxD.md#filter)
+
+## Implementation / Design
+
+### API
+
+#### Select
+
+We will provide a means for clients to enable this feature with a new attribute:
+
+```ts
+export class Select() {
+    ...
+    @attr({ attribute: 'enable-filter', mode: 'boolean' })
+    public enableFilter = false;
```

Comment:
Do you think we'd ever have different filter modes? Some things I can think of configuring are case sensitivity and whether the value needs to start with what's typed in.

If we think there are valid future filter modes, we should consider making this an enum called something like `filterMode` that could have a default value of `none`.